pkg/resource/replication_group/delta_util.go

```diff
@@ -37,6 +40,21 @@ func filterDelta(
 		}
 		// TODO: handle the case of a nil difference (especially when desired EV is nil)
 	}
+
+	// if server has given PreferredMaintenanceWindow a default value, no action needs to be taken
+	if delta.DifferentAt("Spec.PreferredMaintenanceWindow") {
+		if desired.ko.Spec.PreferredMaintenanceWindow == nil && latest.ko.Spec.PreferredMaintenanceWindow != nil {
+			common.RemoveFromDelta(delta, "Spec.PreferredMaintenanceWindow")
+		}
+	}
```

Comment:
Rather than removing from delta after the fact, for default values, I've used the pattern of adding the default value to the object before the delta. See https://github.com/aws-controllers-k8s/s3-controller/blob/f4df629190ec0de88dfe92e22256490ce0775d06/pkg/resource/bucket/hook.go#L361-L372